# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../libxfcegui4/toolbar-fix.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---
diff -Nur libxfcegui4-4.2.2.orig/libxfcegui4/netk-tasklist.c libxfcegui4-4.2.2/libxfcegui4/netk-tasklist.c
--- libxfcegui4-4.2.2.orig/libxfcegui4/netk-tasklist.c	2005-05-15 10:53:29.000000000 +0300
+++ libxfcegui4-4.2.2/libxfcegui4/netk-tasklist.c	2005-10-20 20:59:11.000000000 +0300
@@ -2237,9 +2237,11 @@
 
     if (task->tasklist->priv->show_label) {
         task->label = gtk_label_new (text);
+	gtk_label_set_width_chars(task->label, 1000);
+	gtk_label_set_ellipsize(task->label, PANGO_ELLIPSIZE_END);
         gtk_widget_show (task->label);
         gtk_table_attach (GTK_TABLE (table),
-                      task->label, 1, 2, 0, 1, GTK_EXPAND, GTK_EXPAND, 0, 0);
+                      task->label, 1, 2, 0, 1, GTK_EXPAND|GTK_FILL|GTK_SHRINK, GTK_EXPAND, 0, 0);
     }
 
     gtk_container_add (GTK_CONTAINER (task->button), table);
